packages/dashboards/src/lib/configurator/components/widgets/table/columns-editor-v2/table-columns-configuration-v2.component.html

```diff
@@ -0,0 +1,85 @@
+                    {{ columnForm?.value.label | nuiWidgetConfiguratorSectionHeader : i : "Column" }} <!-- TODO: i18n -->
+                </span>
+                </ng-template>
+
+                <ng-template #headerButtons>
+                    <div class="d-flex">
+                        <!-- TODO: remove "kpi" prefix in v11 -->
+                        <button id="table-widget-remove-column-btn-{{i}}"
+                                nui-button
+                                type="button"
+                                icon="delete"
+                                displayStyle="action"
+                                (click)="removeItem(i)"></button>
+                    </div>
+                </ng-template>
+
+                <nui-widget-configurator-section class="no-margin"
+                                                 [headerTextTemplate]="headerText"
+                                                 [headerButtonsTemplate]="headerButtons">
+                    <nui-table-column-configuration [formControl]="columnForm"
+                                                    [isWidthMessageDisplayed]="isWidthMessageDisplayed"></nui-table-column-configuration>
+                </nui-widget-configurator-section>
+            </div>
+
+            <div class="pl-4 pr-4" *ngIf="(emptyColumns$ | async) && form.touched">
+                <nui-message type="critical" [allowDismiss]="false" i18n>
+                    <b>At least one column must be added.</b>
+                </nui-message>
+            </div>
+
+            <button nui-button
+                    icon="add"
+                    displayStyle="action"
+                    id="table-widget-add-indicator-btn"
+                    type="button"
+                    class="ml-4 mb-4"
+                    (click)="addColumn()"
+                    i18n>Add new column
+            </button>
+
+            <button nui-button
+                    icon="reset"
+                    displayStyle="action"
+                    id="table-widget-reset-indicator-btn"
+                    type="button"
+                    class="ml-4 mb-4"
+                    (click)="onResetColumns()"
+                    i18n>Reset columns
+            </button>
+        </div>
+    </div>
+    <nui-message class="m-4 d-block"
+                 *ngIf="dataSourceManager?.dataSource && !(dataSourceManager?.dataSource?.busy | async) && !dataSourceFields.length"
+                 type="info"
+                 [allowDismiss]="false" i18n>
+        <b>Data source needs to be configured</b>
+    </nui-message>
+    <nui-message class="m-4 d-block" *ngIf="!dataSourceManager?.dataSource" type="info"
+                 [allowDismiss]="false" i18n>
+        <b>Select a data source</b>
+    </nui-message>
+    <nui-message class="m-4 d-block" *ngIf="dataSourceError" type="critical"
+                 [allowDismiss]="false" i18n>
+        <b>The data source errored out with error: </b> {{dataSourceError.type}}
```

Comment:
Maybe "Something went wrong with the data source:"?

"errored out with error" sounds too repetitive to me.